plugins/inputs/hugepages/hugepages_test.go

```diff
@@ -0,0 +1,207 @@
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "2048"}, "nr_hugepages", 2048))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "2048"}, "nr_overcommit_hugepages", 0))
+
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "free_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "resv_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "surplus_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "nr_hugepages_mempolicy", 8))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "nr_hugepages", 8))
+		require.True(t, acc.HasPoint("hugepages_root", map[string]string{"hugepages_size_kb": "1048576"}, "nr_overcommit_hugepages", 0))
+	})
+
+	t.Run("when per node hugepages type is enabled then gather all per node metrics successfully", func(t *testing.T) {
+		h := Hugepages{
+			numaNodePath:  "./testdata/valid/node",
+			gatherPerNode: true,
+		}
+
+		acc := &testutil.Accumulator{}
+		require.NoError(t, h.Gather(acc))
+
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "0"}, "free_hugepages", 434))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "0"}, "surplus_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "0"}, "nr_hugepages", 1024))
+
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "1"}, "free_hugepages", 449))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "1"}, "surplus_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "2048", "node": "1"}, "nr_hugepages", 1024))
+
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "0"}, "free_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "0"}, "surplus_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "0"}, "nr_hugepages", 4))
+
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "1"}, "free_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "1"}, "surplus_hugepages", 0))
+		require.True(t, acc.HasPoint("hugepages_per_node", map[string]string{"hugepages_size_kb": "1048576", "node": "1"}, "nr_hugepages", 4))
```

Comment:
You could construct an `expected` metric table/array and compare the metrics gathered...

Reply:
Changed to `expected` everywhere.